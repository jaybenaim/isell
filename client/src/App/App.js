import React, { Component } from "react";
import "./App.css";
import Nav from "../Nav/Nav";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Products from "../Products/Products";
import { createBrowserHistory } from "history";
import Home from "../Home/Home";
import ProductShow from "../Products/ProductShow";
import Cart from "../Cart/Cart";
import "./App.css";
import Product from "../Data/productSchema";

const history = createBrowserHistory();
class App extends Component {
  state = {
    cartQty: 0,
    cartItems: [],
    selectedProduct: null
  };
  addToCart = (qty, item) => {
    const { id, name, description, price, image } = item;

    const items = [];
    for (let i = 1; i <= qty; i++) {
      items.push(new Product(id + 1, name, description, price, image));
    }
    if (qty === 0 || qty === undefined || qty === "0") {
      qty = 1;
    }
    this.setState(prevState => {
      return {
        cartQty: (prevState.cartQty += qty),
        cartItems: [...prevState.cartItems, ...items]
      };
    });
  };
  setSelectedProduct = product => {
    this.setState({ selectProduct: product });
  };
  render() {
    const { cartItems, cartQty } = this.state;
    return (
      <Router basename="/isell" history={history}>
        <div className="App">
          <Nav cart={{ qty: cartQty, items: cartItems }} />

          <div className="content">
            <Switch>
              <Route exact path="/">
                <Home addToCart={this.addToCart} />
              </Route>
              <Route exact path="/Products">
                <Products
                  addToCart={this.addToCart}
                  selectProduct={this.setSelectedProduct}
                  history={history}
                />
              </Route>
              <Route
                exact
                path="/Products/:id/Show"
                component={ProductShow}
                render={props => (
                  <ProductShow {...props} addToCart={this.addToCart} />
                )}
              />
              <Route
                exact
                path="/ShoppingCart"
                component={Cart}
                render={props => <Cart {...props} />}
              />
            </Switch>
          </div>
          {/* <!-- Footer --> */}
          <footer className="py-5 bg-dark">
            <div className="container">
              <p className="m-0 text-center text-white">
                Copyright &copy; Your Website 2019 NOT A REAL ECOMMERCE SITE DO
                NOT TRY TO PURCHASE
              </p>
            </div>
            {/* <!-- /.container --> */}
          </footer>
        </div>
      </Router>
    );
  }
}

export default App;
