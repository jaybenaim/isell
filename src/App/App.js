import React, { Component } from "react";
import "./App.css";
import Nav from "../Nav/Nav";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import Products from "../Products/Products";
import { createBrowserHistory } from "history";
import Home from "../Home/Home";
import ProductShow from "../Products/ProductShow";
import Cart from "../Cart/Cart";
import Login from "../Register/Login";
import Signup from "../Register/Signup";
import "./App.css";
import Product from "../Data/productSchema";
import Cookies from "js-cookie";
import ProtectedRoute from "../Register/ProtectedRoute";
import ProfileForm from "../Profile/ProfileForm";
const history = createBrowserHistory();
const { token } = Cookies.get();
class App extends Component {
  state = {
    isLoggedIn: token === undefined ? false : true,
    cartQty: 0,
    cartItems: [],
    totalCostBeforeTax: 0,
    selectedProduct: null,
    addedToCart: false,
    valid: false
  };

  calculateTotalBeforeTax = total => {
    this.setState(prevState => {
      return {
        totalCostBeforeTax: (prevState.totalCostBeforeTax += parseFloat(total))
      };
    });
  };
  checkIfItemIsInCart = item => {
    const { cartItems } = this.state;
    let match = false;

    cartItems.map((cartItem, i) => {
      if (item.name === cartItem.name) {
        this.removeFromCart(item.id);
        match = true;
      } else {
        return false;
      }
    });
    if (match) {
      const { showAlert } = this.state;
      this.setState({ showAlert: !showAlert });
    }
  };
  removeFromCart = id => {
    const { cartItems } = this.state;
    const item = cartItems.filter(item => item.id === id);
    this.calculateTotalBeforeTax(-item[0].price);
    this.setState(prevState => {
      return {
        cartItems: [
          ...prevState.cartItems.filter(item =>
            item.id !== id ? item.id : null
          )
        ],
        cartQty: (prevState.cartQty -= prevState.cartQty >= 1 ? 1 : 0)
      };
    });
  };
  addToCart = (qty, item) => {
    this.checkIfItemIsInCart(item);
    const { addedToCart } = this.state;
    const { id, name, description, price, image } = item;
    const items = [];

    for (let i = 1; i <= qty; i++) {
      this.calculateTotalBeforeTax(price);
    }

    items.push(new Product(id, name, description, price, image, "", qty));
    this.setState(prevState => {
      return {
        cartQty: (prevState.cartQty += qty),
        cartItems: [...prevState.cartItems, ...items],
        addedToCart: !addedToCart
      };
    });
  };
  setSelectedProduct = product => {
    this.setState({ selectProduct: product });
  };
  toggleAlert = () => {
    const { showAlert } = this.state;
    this.setState({ showAlert: !showAlert });
  };
  handleLogin = token => {
    this.setState({
      isLoggedIn: token ? true : false
    });
  };

  componentDidUpdate() {}
  render() {
    const {
      cartItems,
      cartQty,
      totalCostBeforeTax,
      showAlert,
      isLoggedIn
    } = this.state;
    return (
      <Router basename="/isell" history={history}>
        <div className="App">
          <Nav
            cart={{ qty: cartQty, items: cartItems }}
            totalCostBeforeTax={totalCostBeforeTax}
            removeFromCart={this.removeFromCart}
            isLoggedIn={isLoggedIn}
            handleLogin={this.handleLogin}
          />

          {showAlert && (
            <div className="alert alert-danger" role="alert">
              <button
                className="btn-danger alert-close-btn"
                onClick={() => this.toggleAlert()}
              >
                X
              </button>
              <br />
              YOU ALREADY ADDED THIS ITEM TO YOUR CART!
              <br />
              <em>Check your cart for multiples</em>
              <br />
            </div>
          )}
          <Link to={{ pathname: "/profiles/new", state: { text: "hello" } }}>
            NEW PROFILE
          </Link>
          <div className="content">
            {/* <button onClick={() => this.checkForUser()}>
              Click to see my secret.
            </button> */}
            <Switch>
              <Route exact path="/">
                <Home addToCart={this.addToCart} isLoggedIn={isLoggedIn} />
              </Route>
              <Route
                exact
                path="/login"
                render={props => (
                  <Login {...props} handleLogin={this.handleLogin} />
                )}
              />
              <Route
                exact
                path="/signup"
                render={props => (
                  <Signup {...props} handleLogin={this.handleLogin} />
                )}
              />

              <Route
                exact
                path="/Products"
                component={Products}
                render={props => (
                  <Products
                    {...props}
                    addToCart={this.addToCart}
                    removeFromCart={this.removeFromCart}
                    selectProduct={this.setSelectedProduct}
                  />
                )}
              />

              <Route
                exact
                path="/Products/:id/Show"
                component={ProductShow}
                render={props => (
                  <ProductShow {...props} addToCart={this.addToCart} />
                )}
              />

              <ProtectedRoute path="/ShoppingCart" isLoggedIn={isLoggedIn}>
                <Route
                  exact
                  path="/ShoppingCart"
                  component={Cart}
                  render={props => (
                    <Cart {...props} removeFromCart={this.removeFromCart} />
                  )}
                />
              </ProtectedRoute>
              <ProtectedRoute path="/profiles/new" isLoggedIn={isLoggedIn}>
                <Route
                  exact
                  path="/profiles/new"
                  render={props => <ProfileForm {...props} />}
                />
              </ProtectedRoute>
            </Switch>
          </div>
          {/* <!-- Footer --> */}
          <footer className="py-5 bg-dark">
            <div className="container">
              <p className="m-0 text-center text-white">
                Copyright &copy; Your Website 2019 NOT A REAL ECOMMERCE SITE DO
                NOT TRY TO PURCHASE
              </p>
            </div>
            {/* <!-- /.container --> */}
          </footer>
        </div>
      </Router>
    );
  }
}

export default App;
