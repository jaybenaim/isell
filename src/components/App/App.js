import React, { Component } from "react";
import "./App.css";
import Nav from "../Nav/Nav";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import { Redirect } from "react-router-dom";
import Products from "../Products/Products";
import { createBrowserHistory } from "history";
import Home from "../Home/Home";
import ProductShow from "../Products/ProductShow";
import Cart from "../Cart/Cart";
import Login from "../Register/Login";
import Signup from "../Register/Signup";
import "./App.css";
import Cookies from "js-cookie";
import ProtectedRoute from "../Register/ProtectedRoute";
import ProfileForm from "../Profile/ProfileForm";
import local from "../../Api/local";
import { connect } from "react-redux";
import { createCart, getCart } from "../../redux/actions";

const history = createBrowserHistory();
const { token } = Cookies.get();

class App extends Component {
  state = {
    isLoggedIn: token === undefined ? false : true,
    selectedProduct: null,
    addedToCart: false,
    valid: false,
    showAlert: false
  };

  setSelectedProduct = product => {
    this.setState({ selectProduct: product });
  };
  toggleAlert = () => {
    const { showAlert } = this.state;
    this.setState({ showAlert: !showAlert });
  };
  handleLogin = token => {
    this.setState({
      isLoggedIn: token ? true : false
    });
  };
  getCart = async id => {
    await local
      .get(`/carts/find/${id}`)
      .then(res => {
        this.props.getCart(res.data);
      })
      .catch(err => {
        alert("Error getting cart", err);
      });
  };
  componentDidMount() {
    const id = Cookies.get("id");
    this.getCart(id);
  }
  render() {
    const { showAlert, isLoggedIn } = this.state;

    return (
      <Router basename="/isell" history={history}>
        <div className="App">
          <Nav isLoggedIn={isLoggedIn} handleLogin={this.handleLogin} />

          {showAlert && (
            <div className="alert alert-danger" role="alert">
              <button
                className="btn-danger alert-close-btn"
                onClick={() => this.toggleAlert()}
              >
                X
              </button>
              <br />
              YOU ALREADY ADDED THIS ITEM TO YOUR CART!
              <br />
              <em>Check your cart for multiples</em>
              <br />
            </div>
          )}
          {/* <Link to={{ pathname: "/profiles/new", state: { text: "hello" } }}>
            NEW PROFILE
          </Link> */}
          <div className="content">
            <Switch>
              <Route exact path="/">
                {/* grab add from redux in home */}
                <Home isLoggedIn={isLoggedIn} />
              </Route>
              <Route
                exact
                path="/login"
                render={props => (
                  <Login {...props} handleLogin={this.handleLogin} />
                )}
              />
              <Route
                exact
                path="/signup"
                render={props => (
                  <Signup {...props} handleLogin={this.handleLogin} />
                )}
              />

              <Route
                exact
                path="/Products"
                render={props => (
                  <Products
                    {...props}
                    removeFromCart={this.removeFromCart}
                    selectProduct={this.setSelectedProduct}
                  />
                )}
              />

              <Route
                exact
                path="/Products/:id/Show"
                component={ProductShow}
                render={props => <ProductShow {...props} />}
              />

              <ProtectedRoute path="/ShoppingCart" isLoggedIn={isLoggedIn}>
                <Route
                  exact
                  path="/ShoppingCart"
                  component={Cart}
                  render={props => (
                    <Cart
                      {...props}
                      timestamp={new Date().toString()}
                      removeFromCart={this.removeFromCart}
                      // items={items}
                    />
                  )}
                />
              </ProtectedRoute>
              <ProtectedRoute path="/profiles/new" isLoggedIn={isLoggedIn}>
                <Route
                  exact
                  path="/profiles/new"
                  render={props => <ProfileForm {...props} />}
                />
              </ProtectedRoute>
            </Switch>
          </div>
          {/* <!-- Footer --> */}
          <footer className="py-5 bg-dark">
            <div className="container">
              <p className="m-0 text-center text-white">
                Copyright &copy; Your Website 2019 NOT A REAL ECOMMERCE SITE DO
                NOT TRY TO PURCHASE
              </p>
            </div>
            {/* <!-- /.container --> */}
          </footer>
        </div>
      </Router>
    );
  }
}
const mapStateToProps = state => {
  const { cart, user } = state.handleItem;

  return { cart, user };
};

export default connect(mapStateToProps, { createCart, getCart })(App);
