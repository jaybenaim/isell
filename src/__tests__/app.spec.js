import React from "react";
import { create } from "react-test-renderer";
import App from "../App/App";
import Cookies from "js-cookie";

describe("app with login and protected routes", () => {
  test("Matches the snapshot", () => {
    const app = create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
  });
});

// STATE checks
describe("isLoggedIn state in app is a boolean and not displaying token", () => {
  test("isLoggedIn is a boolen", () => {
    const cookie = Cookies.get("token");
    const app = create(<App isLoggedIn={cookie} />);
    const instance = app.getInstance();
    expect(typeof instance.state.isLoggedIn === "boolean").toBeTruthy();
  });
});

// calculateTotalCostBeforeTax()
describe("calculateTotalCostBeforTax() sets totalCostBeforeTax to a decimal no greater than 2 points ", () => {
  test("totalCostBeforeTax state from app", () => {
    const totalCostBeforeTax = 34.4444322;
    const finalState = 34.44;
    const app = create(<App totalCostBeforeTax={0} />);
    const instance = app.getInstance();

    instance.calculateTotalBeforeTax(totalCostBeforeTax);

    expect(instance.state.totalCostBeforeTax).toBe(finalState);
  });
});

// addToCart()
describe("add item to state when clicking addToCart", () => {
  test("add one item to cart", () => {
    const app = create(<App cartItems={[]} />);
    const instance = app.getInstance();
    const item = {
      id: 1,
      name: "test",
      price: 4.99,
      image: "https://jaybenaim.github.io/isell/images/img4.jpg",
      description: "test description",
      category: "misc",
      qty: 1
    };
    instance.addToCart(1, item);

    expect(instance.state.cartItems).toStrictEqual(
      expect.objectContaining([item])
    );
  });
});
describe("addToCart increases qty of items", () => {
  test("increase the qty according to the param", () => {
    const app = create(<App cartQty={0} />);
    const instance = app.getInstance();
    const item = {
      id: 1,
      name: "test",
      price: 4.99,
      image: "https://jaybenaim.github.io/isell/images/img4.jpg",
      description: "test description",
      category: "misc",
      qty: 1
    };
    const qty = 4;
    instance.addToCart(qty, item);
    expect(instance.state.cartQty).toStrictEqual(qty);
  });
});

// ALERT TOGGLER
describe("toggleAlert function", () => {
  test("hides alert message", () => {
    const app = create(<App />);
    const instance = app.getInstance();

    instance.toggleAlert(); // show
    instance.toggleAlert(); // hide

    expect(instance.state.showAlert).toBeFalsy();
  });
});
