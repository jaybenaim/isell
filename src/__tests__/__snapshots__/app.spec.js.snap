// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app with login and protected routes Matches the snapshot 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/isell"
    >
      Isell
    </a>
    <button
      className="cart-btn cart-btn-1 btn"
      onClick={[Function]}
    >
      Cart   
      <span>
        0
      </span>
    </button>
    <button
      aria-controls="navbarNavDropdown"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNavDropdown"
      data-toggle="collapse"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNavDropdown"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item active"
          data-target="#navbarNavDropdown"
          data-toggle="collapse"
        >
          <a
            className="nav-link"
            href="/Products"
            onClick={[Function]}
          >
            Products
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
          >
            Pricing
          </a>
        </li>
        <li
          className="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            className="nav-link dropdown-toggle"
            data-toggle="dropdown"
            href="/"
            id="navbarDropdownMenuLink"
          >
            Dropdown link
          </a>
          <div
            aria-labelledby="navbarDropdownMenuLink"
            className="dropdown-menu"
          >
            <a
              className="dropdown-item"
              href="/"
            >
              Action
            </a>
            <a
              className="dropdown-item"
              href="/"
            >
              Another action
            </a>
            <a
              className="dropdown-item"
              href="/"
            >
              Something else here
            </a>
          </div>
        </li>
        <li
          data-target="#navbarNavDropdown"
          data-toggle="collapse"
        >
          <button
            className="cart-btn btn cart-btn-2"
            onClick={[Function]}
          >
            Cart   
            <span>
              0
            </span>
          </button>
        </li>
      </ul>
    </div>
    <div
      className="auth"
    >
      <div
        className="login-container"
      >
        <div
          className="login-btn"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
        <br />
        <div
          className="signup-btn"
        >
          <a
            href="/signup"
            onClick={[Function]}
          >
            Signup
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="content"
  >
    <header>
      <div
        className="carousel slide"
        data-ride="carousel"
        id="home-carousel"
      >
        <ol
          className="carousel-indicators"
        >
          <li
            className="active"
            data-slide-to="0"
            data-target="#home-carousel"
          />
          <li
            data-slide-to="1"
            data-target="#home-carousel"
          />
          <li
            data-slide-to="2"
            data-target="#home-carousel"
          />
        </ol>
        <div
          className="carousel-inner"
          role="listbox"
        >
          <div
            className="carousel-item active carousel-item-1"
            style={
              Object {
                "backgroundImage": "url(\\"https://placehold.it/1900x1080\\")",
              }
            }
          >
            <div
              className="carousel-caption d-none d-md-block"
            >
              <h3>
                50% OFF
              </h3>
              <p
                className="discount"
              >
                Use code SALETIME for 50%
              </p>
            </div>
          </div>
          <div
            className="carousel-item carousel-item-2"
            style={
              Object {
                "backgroundImage": "url(\\"https://placehold.it/1900x1080\\")",
              }
            }
          >
            <div
              className="carousel-caption d-none d-md-block"
            >
              <h3>
                Got a Product?
              </h3>
              <p>
                Let us help you get is sold for top dollar
              </p>
            </div>
          </div>
          <div
            className="carousel-item carousel-item-3"
            style={
              Object {
                "backgroundImage": "url(\\"https://placehold.it/1900x1080\\")",
              }
            }
          >
            <div
              className="carousel-caption d-none d-md-block"
            >
              <h3>
                Local Products
              </h3>
              <p>
                All sellers are in your current area
              </p>
            </div>
          </div>
        </div>
        <a
          className="carousel-control-prev"
          data-slide="prev"
          href="#home-carousel"
          role="button"
        >
          <span
            aria-hidden="true"
            className="carousel-control-prev-icon"
          />
          <span
            className="sr-only"
          >
            Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          data-slide="next"
          href="#home-carousel"
          role="button"
        >
          <span
            aria-hidden="true"
            className="carousel-control-next-icon"
          />
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </div>
    </header>
    <div>
      <div
        className="featured-products-container"
      >
        <div
          className="card text-center"
        >
          <div
            className="card-header"
          >
            <h5
              className="card-title"
            >
              Featured Products
            </h5>
            <p
              className="card-text"
            >
              Take an additional 10% when using the code
               
              <span
                className="text-secondary"
              >
                FREESTUFF 
              </span>
              <br />
            </p>
          </div>
          <div
            className="card-body"
          >
            <div
              className="featured-products-grid"
            >
              <div
                className="card-1"
              >
                <div
                  className="product-card-10001 product-card card"
                >
                  <img
                    alt="none"
                    className="card-img-top"
                    src="https://jaybenaim.github.io/isell/images/img1.jpg"
                  />
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Gucci
                    </h5>
                    <p
                      className="card-text"
                      onClick={[Function]}
                    >
                      Juul Sticker
                    </p>
                    <p>
                      CAD $
                      2.00
                    </p>
                    <label
                      htmlFor="qty-integer"
                    >
                      Qty: 
                    </label>
                    <input
                      className="qty-input"
                      max="10"
                      min="1"
                      name="quantity"
                      onChange={[Function]}
                      placeholder="1"
                      type="number"
                    />
                    <button
                      className="add-to-cart-btn"
                      onClick={[Function]}
                    >
                      Add to cart
                    </button>
                    <a
                      className="btn btn-primary"
                      href="/Products/10001/show"
                      onClick={[Function]}
                    >
                      Show Details
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="card-2"
              >
                <div
                  className="product-card-10002 product-card card"
                >
                  <img
                    alt="none"
                    className="card-img-top"
                    src="https://jaybenaim.github.io/isell/images/img2.jpg"
                  />
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Rick and Morty
                    </h5>
                    <p
                      className="card-text"
                      onClick={[Function]}
                    >
                      Juul Sticker
                    </p>
                    <p>
                      CAD $
                      2.00
                    </p>
                    <label
                      htmlFor="qty-integer"
                    >
                      Qty: 
                    </label>
                    <input
                      className="qty-input"
                      max="10"
                      min="1"
                      name="quantity"
                      onChange={[Function]}
                      placeholder="1"
                      type="number"
                    />
                    <button
                      className="add-to-cart-btn"
                      onClick={[Function]}
                    >
                      Add to cart
                    </button>
                    <a
                      className="btn btn-primary"
                      href="/Products/10002/show"
                      onClick={[Function]}
                    >
                      Show Details
                    </a>
                  </div>
                </div>
              </div>
              <div
                className="card-3"
              >
                <div
                  className="product-card-10003 product-card card"
                >
                  <img
                    alt="none"
                    className="card-img-top"
                    src="https://jaybenaim.github.io/isell/images/img3.jpg"
                  />
                  <div
                    className="card-body"
                  >
                    <h5
                      className="card-title"
                    >
                      Supreme
                    </h5>
                    <p
                      className="card-text"
                      onClick={[Function]}
                    >
                      Juul Sticker
                    </p>
                    <p>
                      CAD $
                      2.00
                    </p>
                    <label
                      htmlFor="qty-integer"
                    >
                      Qty: 
                    </label>
                    <input
                      className="qty-input"
                      max="10"
                      min="1"
                      name="quantity"
                      onChange={[Function]}
                      placeholder="1"
                      type="number"
                    />
                    <button
                      className="add-to-cart-btn"
                      onClick={[Function]}
                    >
                      Add to cart
                    </button>
                    <a
                      className="btn btn-primary"
                      href="/Products/10003/show"
                      onClick={[Function]}
                    >
                      Show Details
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card-footer text-muted"
          >
            2 days ago
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="py-5 bg-dark"
  >
    <div
      className="container"
    >
      <p
        className="m-0 text-center text-white"
      >
        Copyright © Your Website 2019 NOT A REAL ECOMMERCE SITE DO NOT TRY TO PURCHASE
      </p>
    </div>
  </footer>
</div>
`;
